### DISCUSSION TITLE
glTF animation import option to not change any animation channels and keys.

### DISCUSSION THREAD

**Discussion by emperorofmars** [godotengine/godot-proposals#6218] [Editor]
Tested everything with: Godot 4.0 Beta 16

If i import a glTF 2.0 file with animations, i have the option in the "Advanced Import Settings" window on the AnimationPlayer to set "Import Tracks" to "IfPresentForAll".
In that case additional channels and keyframes are set on most animations. I assume this is indended behaviour?

If i set "Import Tracks" to "IfPresent", all channels and keys which don't differ from the imported pose are omitted.
For example i have an animation called 'Default' made already, which will be empty.
(As an aside: i would like also to be able to designate one of the imported animations as the RESET.)

I would like to import a gltf file, edit its animations and other properties, and export it again. Currently i would have to put in a lot of effort to manually clean it up with either setting.

An option to import animations exactly as they are, without adding or removing channels and keys, would be optimal.

This works fine in Godot 3, Unity 2019 with UnityGLTF, as well as importing the glTF file back into Blender 3.4.1.

Edit: Reworded some sentences for better clarity and added minimal example.

Minimal example .glb:
[bugreport_minimal_example_glb.zip](https://github.com/godotengine/godot-proposals/files/10560268/bugreport_minimal_example_glb.zip)

This had an animation called 'Default'. On import with "Import Tracks" set to "IfPresent" it will be empty.
It has two animations which animate only the arms, not the rest of the body. On import with "Import Tracks" set to "IfPresentForAll", keys for all bones are set.
Import into Blender 3.4.1 and view the Actions to see the original.

**Comment by Calinou** [comment]
cc @fire

**Reply by fire** [reply]
I think this is a bug that external animation are not saved, but will need to help to write the recreation steps as it's unclear to me.

**Reply by emperorofmars** [reply]
I reworded the original post for better clarity and added a minimal example .glb file.

**Comment by emperorofmars** [comment]
In the 4.0 release there is an import option called 'remove_immutable_tracks'. Toggling that off resolves this issue.
Thank you to the developers!
